lib/resty/kafka/producer.lua

```diff
@@ -353,12 +355,82 @@ local function negotiate_api_version(supported_version_map, api_key, max_support
     return min_version
 end
 
+-- if new arguments already exist in cached producers, then skip creating
+-- may subject to plugin (e.g. "kafka-log") schema updates
+local function _same_cluster(broker_list, producer_config, cluster_name)
+  local same_cluster = true
+
+  local producer = cluster_inited[cluster_name]
+  local producer_client = producer.client
+  local producer_ringbuffer = producer.ringbuffer
+
+  -- compare producer config
+  for k1, v1 in pairs(producer_config) do
+    -- values in producer
+    local pkeys = { "required_acks", "request_timeout", "batch_num", "batch_size",
+                    "max_retry", "retry_backoff", "flush_time" }
+    if tx_find(pkeys, k1) and producer[k1] ~= v1 then
+      same_cluster = false
+      break
+    end
+
+    if k1 == "producer_type" then
+      if v1 == "async" and not producer["async"] then
+        same_cluster = false
+        break
+      end
+
+      if v1 == "sync" and producer["async"] then
+        same_cluster = false
+        break
+      end
+    end
+
+    if k1 == "auth_config" and not tx_deepcompare(producer[k1], v1, true) then
+      same_cluster = false
+      break
+    end
+
+    -- values in client
+    local ckeys = { "socket_timeout", "keepalive_timeout", "ssl", "client_cert", "client_priv_key" }
+    if tx_find(ckeys, k1) and producer_client["socket_config"][k1] ~= v1 then
+      same_cluster = false
+      break
+    end
+
+    -- values in ringbuffer
+    if k1 == "max_buffering" and producer_ringbuffer["size"] ~= v1 * 3 then
+      same_cluster = false
+      break
+    end
+  end
+
+  -- compare broker list - the core elements
+  if not tx_deepcompare(broker_list, producer_client.broker_list, true) then
```

Comment:
Just a thought of mine. What if the same list with different ordering?

Reply:
Done.